content/en/docs/concepts/storage/storage-classes.md

```diff
@@ -172,23 +193,21 @@ and [taints and tolerations](/docs/concepts/scheduling-eviction/taint-and-tolera
 
 The following plugins support `WaitForFirstConsumer` with dynamic provisioning:
 
+- CSI volumes, provided that the specific CSI driver supports this
```

Comment:
This list is woefully out of date. Maybe best to remove GCEPersistentDisk and just mention CSI (and Local below)

Reply:
Ideally, we can get a current list of in-tree providers that do provide that support. If it's in-tree, we try to document it _somewhere_. Even if it's also deprecated.

Reply:
Off the top of my head, we have local, gce-pd, aws-ebs, azure-disk, azure-file. Maybe vsphere and ceph? @jsafrane can you confirm?

Reply:
The `awsElasticBlockStore` and `azureDisk` volume types are not included in Kubernetes v1.28 - is that right?

Reply:
In 1.29, all cloud-based volume plugins are migrated to CSI, so just `local` remains in-tree and supports `WaitForFirstConsumer`. So CSI and `local` is more appropriate.

Reply:
Bear in mind: these are the docs for v1.28

Reply:
IMO it'd be nice to get this page updated ahead of the v1.29 release.

Reply:
(v1.29 is now released, and I've rebased)

Reply:
Is there any feature gate that lets you re-enable in tree support for azure file? I think we may have some work to do, as a project, to make sure the docs are up to date around those removals.